addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -67,8 +67,8 @@ export class Attachment extends FileModelMixin(Record) {
     }
 
     get isDeletable() {
-        if (this.message && this.store.self.main_user_id?.share !== false) {
-            return this.message.editable;
+        if (!!this.message_ids.length && this.store.self.main_user_id?.share !== false) {
+            return this.this.message_ids[0].editable;
```

Comment:
I'm not sure about using [0], usually handling a "list" should be done with either "all" or "any" depending on what we need.